src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -20277,6 +20280,804 @@ static RefByteContainer M42(scoped ref ByteContainer bc)
+                // (9,15): error CS8936: Feature 'ref fields' is not available in C# 10.0. Please use language version 11.0 or greater.
+                //         using scoped ref readonly R r5 = ref r;
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion10, "scoped").WithArguments("ref fields", "11.0").WithLocation(9, 15),
+                // (9,22): error CS1073: Unexpected token 'ref'
+                //         using scoped ref readonly R r5 = ref r;
+                Diagnostic(ErrorCode.ERR_UnexpectedToken, "ref").WithArguments("ref").WithLocation(9, 22)
+                );
+            verify(comp);
+
+            comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (8,22): error CS1073: Unexpected token 'ref'
+                //         using scoped ref R r2 = ref r;
+                Diagnostic(ErrorCode.ERR_UnexpectedToken, "ref").WithArguments("ref").WithLocation(8, 22),
+                // (9,22): error CS1073: Unexpected token 'ref'
+                //         using scoped ref readonly R r5 = ref r;
+                Diagnostic(ErrorCode.ERR_UnexpectedToken, "ref").WithArguments("ref").WithLocation(9, 22)
+                );
+            verify(comp);
+
+            static void verify(CSharpCompilation comp)
+            {
+                var tree = comp.SyntaxTrees[0];
+                var model = comp.GetSemanticModel(tree);
+                var decls = tree.GetRoot().DescendantNodes().OfType<VariableDeclaratorSyntax>().ToArray();
+                var locals = decls.Select(d => model.GetDeclaredSymbol(d).GetSymbol<LocalSymbol>()).ToArray();
+
+                VerifyLocalSymbol(locals[0], "scoped R r1", RefKind.None, DeclarationScope.ValueScoped);
+                VerifyLocalSymbol(locals[1], "scoped ref R r2", RefKind.Ref, DeclarationScope.RefScoped);
+                VerifyLocalSymbol(locals[2], "scoped ref readonly R r5", RefKind.RefReadOnly, DeclarationScope.RefScoped);
+
+                foreach (var decl in decls)
+                {
+                    var type = ((VariableDeclarationSyntax)decl.Parent).Type;
+                    Assert.Null(model.GetTypeInfo(type).Type);
+                    Assert.Equal("R", model.GetSymbolInfo(type.SkipScoped(out _).SkipRef(out _)).Symbol.ToTestDisplayString());
+                    Assert.True(SyntaxFacts.IsInTypeOnlyContext(type));
+                    Assert.True(SyntaxFacts.IsInTypeOnlyContext(type.SkipScoped(out _)));
+                    Assert.True(SyntaxFacts.IsInTypeOnlyContext(type.SkipScoped(out _).SkipRef(out _)));
+                }
+            }
+        }
+
+        [Fact]
+        public void LocalScope_01_UsingStmt()
+        {
+            var source =
+@"#pragma warning disable 219
+
+class Program
+{
+    static void F(ref R r)
+    {
+        using (scoped R r1 = default) {}
+        using (scoped ref R r2 = ref r) {}
+        using (scoped ref readonly R r5 = ref r) {}
```

Comment:
>using (scoped ref readonly R r5 = ref r) {} [](http://example.com/codeflow?start=8&length=43)

Do we have any binding tests for `using` declarations and statements with `ref` without `scoped`? It looks like parsing succeeds in those cases and we expect binding to report an error but it doesn't look like we have tests that confirm that.
```csharp
static void F(ref R r1)
{
    using ref R r2 = ref r1;
    using ref readonly R r3 = ref r1;
    using (ref R r4 = ref r1) { }
    using (ref readonly R r5 = ref r1) { }
}
```

Reply:
> It looks like parsing succeeds in those cases and we expect binding to report an error but it doesn't look like we have tests that confirm that.

This is how it works, i.e. a binding error is reported. Creating tests for legacy scenarios is out of scope of this PR.

Reply:
It is the same error as the one reported with scoped:
```
// (8,23): error CS1073: Unexpected token 'ref
```
And reported by the same existing code.

Reply:
> It is the same error as the one reported with scoped:

I understand, but I couldn't find any tests that confirmed that.

Reply:
> I understand, but I couldn't find any tests that confirmed that.

I am sure there are other ways to confirm this legacy behavior.

Reply:
If the confirmation is what you are after, the behavior can be observed here [https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgNwgCcACUgXhJgFMB3EgJQAoBKAbgFgAoAAQAYSTItQBmjMgCYSlYWKIsSAbwC+3bmiIBXAMYJxIEjwCMAOgCSAESgoADgHsUEYABtq3RdxJfDAZkMAWEitbB2pWJVUuZSA=](https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgNwgCcACUgXhJgFMB3EgJQAoBKAbgFgAoAAQAYSTItQBmjMgCYSlYWKIsSAbwC+3bmiIBXAMYJxIEjwCMAOgCSAESgoADgHsUEYABtq3RdxJfDAZkMAWEitbB2pWJVUuZSA=)